### DISCUSSION TITLE
JSON config for Sonoff T1 2CH EU

### DISCUSSION THREAD

**Discussion by RuudSchellekens** [RavenSystem/esp-homekit-devices#1701] [MEPLHAA Scripts]
I have flashed the HAA firmware to the Sonoff T1 2ch EU wall switch (the web interface shows HAA v9.5.1 griffon)0. Wifi config works well but the button config does not. 

I want to use it as 2 buttons for 2 lights to turn them on and off. If I use the switch 2 ch config from the device database, non of the buttons work at all. I tried both grouped and different room configs > no working switch. 

I think it works slightly different as the second button seems to be connected to a second chip. I would like to test and figure this out but I need some help since documentation etc is very sparse if I google. Any help will be appreciated.

**Comment by RavenSystem** [comment]
This JSON config should work, based on this info: https://templates.blakadder.com/sonoff_T1_eu_2.html

```json
{"c":{"l":13},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0,"t":0}],"es":[{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9,"t":0}]}]}]}
```

**Reply by RuudSchellekens** [reply]
Thanks for the quick reply. I tested this JSON string and it makes only 1 button work (at least the physical button switch of the second Channel does not trigger at all). I just flashed Tasmota to rule out a hardware issues and that does work with 2 buttons physically and via WiFi. I'll try to debug it a bit more to make both buttons work. Any suggestions are still welcome.

**Reply by RavenSystem** [reply]
It is easy: See how Tasmota is configured.

**Reply by RuudSchellekens** [reply]
It turns out the config works fine in Home. The problem is the left button that sometimes does work and other times not. But via the web interface of Tasmota and via Home it both works fine. Sorry to bother you with this. I'll post the config you made as a new device request so it can be added to the device database. Thanks for your great help and support!
